src/features/Applet/Schedule/Calendar/Calendar.utils.tsx

```diff
@@ -7,61 +7,183 @@ import {
+
 export const formatToYearMonthDate = (date?: Date) =>
   date && format(date, DateFormats.DayMonthYear);
 
+export const formatToWeekYear = (date: Date) => `${getISOWeek(date)} ${date.getFullYear()}`;
+
+export const allDayEventsSortedByDays = mockedEvents
+  .reduce(
+    (
+      acc: { date: string | undefined; eventsIds: { id: string; isHidden: boolean }[] }[],
+      event,
+    ) => {
+      const currentEventStartDate = formatToYearMonthDate(event.start);
+      if (event.allDayEvent || event.alwaysAvailable) {
+        if (acc.some((el) => el.date === currentEventStartDate)) {
+          acc.map((el) => {
+            if (el.date === currentEventStartDate) {
+              const ids = el.eventsIds;
+
+              return {
+                ...el,
+                eventsIds: ids.push({
+                  id: event.id,
+                  isHidden: ids.length > LENGTH_TO_SET_ID_IS_HIDDEN,
+                }),
+              };
+            }
+
+            return el;
+          });
+        } else {
+          acc.push({
+            date: currentEventStartDate,
+            eventsIds: [{ id: event.id, isHidden: false }],
+          });
+        }
+      }
+
+      return acc;
+    },
+    [],
+  )
+  .filter((el) => el.eventsIds.length > 4);
+
+export const hiddenEventsIds = allDayEventsSortedByDays.reduce((acc: string[], item) => {
+  item.eventsIds.forEach((el) => el.isHidden && acc.push(el.id));
+
+  return acc;
+}, []);
+
+export const getProcessedEvents = (date: Date) =>
+  mockedEvents.map((event) => ({
+    ...event,
+    isOffRange: event.start.getMonth() !== date.getMonth(),
+    isHiddenInTimeView: hiddenEventsIds.some((id) => id === event.id),
+  }));
+
+export const getHasWrapperMoreBtn = (
+  activeView: CalendarViews,
+  events: CalendarEvent[],
+  date: Date,
+  isAllDayEventsVisible: AllDayEventsVisible,
+) => {
+  const currentDate = formatToYearMonthDate(date);
+  const currentWeek = formatToWeekYear(date);
+  const hasDateHiddenEvents =
+    !isAllDayEventsVisible && allDayEventsSortedByDays.some((item) => item.date === currentDate);
+  const hasDateHiddenEventsWithState =
+    isAllDayEventsVisible?.period === currentDate && !isAllDayEventsVisible?.visible;
+  const hasWeekHiddenEvents =
+    !isAllDayEventsVisible &&
+    allDayEventsSortedByDays.some(
+      (item) => item.date && formatToWeekYear(new Date(item.date)) === currentWeek,
+    );
+  const hasWeekHiddenEventsWithState =
+    isAllDayEventsVisible?.period === currentWeek && !isAllDayEventsVisible?.visible;
+
+  switch (activeView) {
+    case CalendarViews.Day:
+      return hasDateHiddenEvents || hasDateHiddenEventsWithState;
+    case CalendarViews.Week:
+      return hasWeekHiddenEvents || hasWeekHiddenEventsWithState;
+    default:
+      return false;
+  }
+};
+
+export const getDayName = (date: Date) =>
+  date.toLocaleDateString(i18n.language, { weekday: 'long' });
+
+export const getMonthName = (date: Date, length?: 'long' | 'short') =>
```

Comment:
maybe it would be better to move to the enum ('long' | 'short')?

Reply:
done